### PR TITLE
`makeEnrollment`: Use `Height` instead of `cycle`

### PR DESCRIPTION
**Pull request by Geod24**
This will make transitioning from `distance` to `Height` simpler.

### CODE REVIEW

File: source/agora/consensus/EnrollmentManager.d [hunk 1]
```diff
@@ -1137,6 +1134,8 @@ unittest
     // create an EnrollmentManager object
     auto params = new immutable(ConsensusParams)();
     auto man = new EnrollmentManager(key_pair, params);
+    // Useful constant
+    const EnrollAt = Height(1);
```
Comment (hewison-chris) [inline source/agora/consensus/EnrollmentManager.d]: If const then maybe name could include the fact it is for height 1.

File: source/agora/consensus/EnrollmentManager.d [hunk 2]
```diff
@@ -460,15 +460,12 @@ public class EnrollmentManager
 
     /// Ditto
     version (unittest) public static Enrollment makeEnrollment (
-        in Hash utxo, in KeyPair key, uint cycle_length, uint offset = 0)
+        in Hash utxo, in KeyPair key, in Height height, uint cycle_length)
```
Comment (hewison-chris) [inline source/agora/consensus/EnrollmentManager.d:463]: Looks familiar.

File: source/agora/consensus/data/genesis/package.d [hunk 3]
```diff
@@ -91,7 +92,7 @@ version (unittest) public void checkGenesisEnrollments (
     Hash txhash = hashFull(genesisBlock.txs.filter!(tx => tx.type == TxType.Freeze).front);
     Hash[] utxos = iota(6).map!(i => UTXO.getHash(txhash, i)).array;
     auto enrollments = utxos.enumerate.map!(en =>
-        NodeEnrollment(EnrollmentManager.makeEnrollment(en.value, keys[en.index], cycle_length, 0),
+            NodeEnrollment(EnrollmentManager.makeEnrollment(en.value, keys[en.index], Height(0), cycle_length),
```
Comment (hewison-chris) [inline source/agora/consensus/data/genesis/package.d]: Extra spaces?

**Review by hewison-chris — APPROVED** [review]
LGTM just a couple of suggestions.

File: source/agora/consensus/EnrollmentManager.d [hunk 4]
```diff
@@ -460,15 +460,12 @@ public class EnrollmentManager
...
         @trusted nothrow
     {
-        // Generate the commitment to use
-        auto cache = PreImageCache(PreImageCycle.NumberOfCycles, cycle_length);
-        assert(offset < cache.length);
-        cache.reset(hashMulti(key.secret, "consensus.preimages", offset));
-
-        return makeEnrollment(utxo, key, cache[$ - offset - 1], cycle_length, offset);
+        // Generate the random seed to use
+        auto cache = PreImageCycle(key.secret, cycle_length);
```
Comment (linked0) [inline source/agora/consensus/EnrollmentManager.d]: `cycle` would be better instead of `cache`

**Review by linked0 — APPROVED** [review]
LGTM just some nits.
There's a typo in the first commit's message.(`abotu`)

**Comment by Geod24** [comment]
All comments have been addressed.

**Pull request merged**